src/components/UploadAvatarModal/UploadAvatarModal.js

```diff
@@ -0,0 +1,247 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import PropTypes from 'prop-types'
+import Slider from 'rc-slider'
+import { useState, useCallback } from 'react'
+import Cropper from 'react-easy-crop'
+import { useDispatch } from 'react-redux'
+
+import asModal, { ModalContent, ModalFooter } from '~/components/asModal'
+import { updateAvatar } from '~/store/actions/user'
+import getResponseError from '~/util/getResponseError'
+
+import UploadAvatarMessageBox from './UploadAvatarMessageBox'
+import styles from './UploadAvatarModal.module.scss'
+
+// Component Constants
+const MAX_FILE_SIZE = 26214400 // Server upload max is 25M
+const SUBMIT_AUTO_CLOSE_DELAY_TIME = 3000
+const HALF_CIRCLE = 180 // Conversion of rat to deg
+
+
+function UploadAvatarModal (props) {
+  const {
+    onClose,
+    isOpen,
+  } = props
+
+  const [result, setResult] = useState({})
+
+  const [upImg, setUpImg] = useState()
+  const [crop, setCrop] = useState({ x: 0, y: 0 }) // Lint comment on "Identifier name is too short" - this is required by the Cropper. Cannot change.
+  const [zoom, setZoom] = useState(1)
+  const [rotation, setRotation] = useState(0)
+  const [croppedAreaPixels, setCroppedAreaPixels] = useState(null)
+  const [submitReady, setSubmitReady] = useState(false)
+  const [submitting, setSubmitting] = useState(false)
+
+  // Linting note: Cannot use arrow function in onevent, need to wrap
```

Comment:
Same as above, Please do not leave "lint notes" like this. The linter makes this pattern of requiring stable functions consistent across the project, therefore there's little value in pointing that out here.